sn_cli/src/subcommands/files/mod.rs

```diff
@@ -251,6 +258,28 @@ pub(crate) async fn files_cmds(
     Ok(())
 }
 
+/// Estimate the upload cost of a chosen file
+pub(crate) async fn estimate_cost(path: PathBuf, client: &Client, root_dir: &Path) -> Result<()> {
+    let mut chunk_manager = ChunkManager::new(root_dir);
+    chunk_manager.chunk_path(&path, false, false)?;
+
+    let mut estimate: u64 = 0;
+
+    for chunk in chunk_manager.get_chunks() {
```

Comment:
as @RolandSherwin mentioned, we want to push each of these into a task so that this can happen in an async fashion (otherwise this could take quite a while)

Reply:
Okay. If it's okay with you @RolandSherwin I'll be in touch to ask you your thoughts on the best way to do this.

Reply:
Sure, I'm glad to be of help here.  😄